sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -204,11 +214,8 @@ class Analyzer(override val catalogManager: CatalogManager)
   }
 
   override def execute(plan: LogicalPlan): LogicalPlan = {
-    AnalysisContext.reset()
-    try {
+    AnalysisContext.withNewAnalysisContext {
       executeSameContext(plan)
-    } finally {
-      AnalysisContext.reset()
```

Comment:
The rule `ResolveEncodersInUDF` will call `analyzer. execute()` again, the AnalysisContext will be messed up after that.

Reply:
This looks like a bug, can we open a separate PR for it? then we can backport it.